Objects/object.c

```diff
@@ -1712,6 +1712,29 @@ notimplemented_dealloc(PyObject* ignore)
     Py_FatalError("deallocating NotImplemented");
 }
 
+static int
+notimplemented_bool(PyObject *v)
+{
+    if (PyErr_WarnEx(PyExc_DeprecationWarning,
+                     "NotImplemented should not be used in a boolean context",
+                     1) < 0)
+        return -1;
+    return 1;
+}
+
+static PyNumberMethods notimplemented_as_number = {
+    0,                              /* nb_add */
+    0,                              /* nb_subtract */
+    0,                              /* nb_multiply */
+    0,                              /* nb_remainder */
+    0,                              /* nb_divmod */
+    0,                              /* nb_power */
+    0,                              /* nb_negative */
+    0,                              /* nb_positive */
+    0,                              /* nb_absolute */
+    (inquiry)notimplemented_bool,   /* nb_bool */
```

Comment:
`(inquiry)` is not needed.
You can use also more short form:
```c
static PyNumberMethods notimplemented_as_number = {
   .nb_bool = notimplemented_bool
};
```

Reply:
The inquiry cast is because I was just copy pasting from None's number methods and then modifying the name and None (equally unnecessarily it seems) also had the cast. I didn't notice the PEP7 update that allowed C99 designated initializers; I'll definitely change to use that.

Reply:
Done. I left a trailing comma after notimplemented_bool to reduce diff sizes if it ever grows another number method for who knows what reason, but otherwise it's identical to your suggestion.